nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/ParameterContextResource.java

```diff
@@ -323,6 +356,204 @@ public Response updateParameterContext(
+        }
+        if (assetContents == null) {
+            throw new IllegalArgumentException("Asset contents must be specified.");
+        }
+
+        // If clustered and not all nodes are connected, do not allow creating an asset.
+        // Generally, we allow the flow to be modified when nodes are disconnected, but we do not allow creating an asset because
+        // the cluster has no mechanism for synchronizing those assets after the upload.
+        final ClusterCoordinator clusterCoordinator = getClusterCoordinator();
+        if (clusterCoordinator != null) {
+            final Set<NodeIdentifier> disconnectedNodes = clusterCoordinator.getNodeIdentifiers(NodeConnectionState.CONNECTING, NodeConnectionState.DISCONNECTED, NodeConnectionState.DISCONNECTING);
+            if (!disconnectedNodes.isEmpty()) {
+                throw new IllegalStateException("Cannot create an Asset because the following %s nodes are not currently connected: %s".formatted(disconnectedNodes.size(), disconnectedNodes));
+            }
+        }
+
+        // Get the context or throw ResourceNotFoundException
+        final NiFiUser user = NiFiUserUtils.getNiFiUser();
+        final ParameterContextEntity contextEntity = serviceFacade.getParameterContext(contextId, false, user);
+        final Set<AffectedComponentEntity> affectedComponents = serviceFacade.getComponentsAffectedByParameterContextUpdate(Collections.singletonList(contextEntity.getComponent()));
+
+        // Authorize the request
+        serviceFacade.authorizeAccess(lookup -> {
+            // Verify READ and WRITE permissions for user, for the Parameter Context itself
+            final ParameterContext parameterContext = lookup.getParameterContext(contextId);
+            parameterContext.authorize(authorizer, RequestAction.READ, user);
+            parameterContext.authorize(authorizer, RequestAction.WRITE, user);
+
+            // Verify READ and WRITE permissions for user, for every component that is affected
+            affectedComponents.forEach(component -> parameterUpdateManager.authorizeAffectedComponent(component, lookup, user, true, true));
+        });
+
+        // If we need to replicate the request, we do so using the Upload Request Replicator, rather than the typical replicate() method.
+        // This is because Upload Request Replication works differently in that it needs to be able to replicate the InputStream multiple times,
+        // so it must create a file on disk to do so and then use the file's content to replicate the request. It also bypasses the two-phase
+        // commit process that is used for other requests because doing so would result in uploading the file twice to each node or providing a
+        // different request for each of the two phases.
+
+        final long startTime = System.currentTimeMillis();
+        final InputStream maxLengthInputStream = new MaxLengthInputStream(assetContents, (long) DataUnit.GB.toB(1));
```

Comment:
For tracking purposes, it would be helpful to promote the maximum length value to a static value. Although this could be made configurable, it seems unlikely to change, and 1 GB seems like it should be large enough to handle anything of reasonable size.